app/controllers/users.js

```diff
@@ -29,17 +29,132 @@ exports.authCallback = (req, res) => {
 
 exports.findUsers = (req, res) => {
   User.find({}).select('name email').then((allUsers) => {
-    res.send(allUsers);
+    res.status(200)
+      .json(allUsers);
   });
 };
 
 exports.findUser = (req, res) => {
   const userid = req.params.userid;
   User.findById(userid, (err, oneUser) => {
     if (!err) {
-      res.send(oneUser);
+      res.status(200)
+        .json(oneUser);
     } else {
-      res.send('An error occurred');
+      res.status(400)
+        .json('An error occured');
+    }
+  });
+};
+
+exports.getFriends = (req, res) => {
+  const userId = req.body.userId;
+  User.findById(userId, (err, user) => {
+    if (!err) {
+      res.status(200)
+        .json(user.friends);
```

Comment:
Probably want to check that user is defined if not return a `404`. Mongoose `findById` wouldn't throw an error if the user does not exist.